### PR TITLE
Laboratory work #1, Irina Pereyaslavtseva - 22FPL1

### PR DESCRIPTION
**Pull request by iriper**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1 @@
+print('hello world')
```
Comment (artyomtugaryov) [inline main.py:1]: Remove this file

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Nothing to review. Any way I hope you will done in time.

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -11,6 +11,15 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if isinstance(text, str):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:14]: use early return - https://habr.com/ru/articles/348074/

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -11,6 +11,15 @@ def tokenize(text: str) -> list[str] | None:
...
+        text = text.lower()
+        tokens = []
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:16]: try list comprehension

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Comment by artyomtugaryov** [comment]
Nothing to review today

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -11,6 +11,11 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    else:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:16]: else is not needed

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -11,6 +11,11 @@ def tokenize(text: str) -> list[str] | None:
...
+        tokens = [symbol.lower() for symbol in text if symbol.isalpha()]
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:17]: do not create the variable

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -19,6 +24,15 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list):
+        return None
+    for token in tokens:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:29]: merge two loops, please

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -28,6 +42,11 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
     :param text: a text
     :return: a dictionary with two keys – name, freq
     """
+    if not isinstance(language, str) or not isinstance(text, str):
+        return None
+    dict_of_freq = calculate_frequencies(tokenize(text))
+    lang_prof = {'name': language, 'freq': dict_of_freq}
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:48]: do not create the variable - return as is

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -39,6 +58,7 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
     """
 
 
+
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:61]: revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 4 above)
Comment (iriper) [reply lab_1_classify_by_unigrams/main.py:16]: done

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (iriper) [reply lab_1_classify_by_unigrams/main.py:17]: done

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (iriper) [reply lab_1_classify_by_unigrams/main.py:48]: done

File: lab_1_classify_by_unigrams/main.py (hunk 8 above)
Comment (iriper) [reply lab_1_classify_by_unigrams/main.py]: done

**Pull request closed without merging**